### ISSUE TITLE
No ".reads.gz" file generated

### ISSUE THREAD

**Issue by chlee-tabin** [hms-dbmi/dropEst#25]
I updated to the most recent version of dropEst package (0.8.0) and had successfully compiled it. When I am running the droptag with -s option, I get an extra ".params.gz" file, but not ".reads.gz" file. The documentation specifies I should use ".reads.gz" file for the -r option. Am I missing something?

**Comment by VPetukhov** [comment]
Normally, it must create ".reads.gz". Can you publish full log here? And did you work with previous versions?

**Comment by chlee-tabin** [comment]
Sorry I am not sure what to publish as a full log, so I pasted what I think is relevant.

I have worked with previous version before. I log in to O2 (HMS cluster) and run the compiled binaries:

@compute-e-16-230:~/dropEst$ ./droptag -h
droptag -- generate tagged fastq files for alignment
Version: 0.8.0

SYNOPSIS
	droptag [options] -c config.xml barcode_reads.fastq [barcode_umi_reads.fastq] gene_reads.fastq [library_tags.fastq]
OPTIONS:
	-c, --config filename: xml file with droptag parameters
	-h, --help: show this info
	-l, --log-prefix prefix: logs prefix
	-n, --name name: alternative output base name
	-p, --parallel number: number of threads
	-s, --save-reads-params : serialize reads parameters to save quality info
	-S, --save-stats : save stats to rds file
	-t, --lib-tag library tag : (for IndropV3 with library tag only)
	-q, --quiet : disable logs

@compute-e-16-230:~/dropEst$ ./dropest -h
dropest: estimate molecular counts per cell
Version: 0.8.0

SYNOPSIS
	dropest [options] -c config.xml file_1.bam [..., file_n.bam]
OPTIONS:
	-b, --bam-output: print tagged bam files
	-c, --config filename: xml file with estimation parameters
	-C, --cells num: maximal number of output cells
	-f, --filled-bam: bam file already contains genes/barcodes tags
	-F, --filtered-bam: print tagged bam file after the merge and filtration
	-g, --genes filename: file with genes annotations (.bed or .gtf)
	-G, --genes-min num: minimal number of genes in output cells
	-h, --help: show this info
	-l, --log-prefix : logs prefix
	-L, --gene-match-level :
		e: count UMIs with exonic reads only;
		i: count UMIs with intronic reads only;
		E: count UMIs, which have both exonic and not annotated reads;
		I: count UMIs, which have both intronic and not annotated reads;
		B: count UMIs, which have both exonic and intronic reads;
		A: count UMIs, which have exonic, intronic and not annotated reads.
		Default: -L eEBA.
	-m, --merge-barcodes : merge linked cell tags
	-M, --merge-barcodes-precise : use precise merge strategy (can be slow), recommended to use when the list of real barcodes is not available
	-o, --output-file filename : output file name
	-r, --read-params filenames: file or files with serialized params from tags search step. If there are several files, they should be provided in quotes, separated by space: "file1.reads.gz file2.reads.gz file3.reads.gz"
	-P, --pseudoaligner: use chromosome name as a source of gene id
	-q, --quiet : disable logs
	-R, --reads-output: print count matrix for reads and don't use UMI statistics
	-V, --velocyto : save separate count matrices for exons, introns and exon/intron spanning reads
	-w, --write-mtx : write out matrix in MatrixMarket format

I have run droptag and this is the command I used:

~/dropEst/droptag -c ~/dropEst/configs/indrop_v3_with_barcodes.xml -S -s -l dlib${SLURM_ARRAY_TASK_ID}_FC03554 -n drop/dlib${SLURM_ARRAY_TASK_ID}.FC03554.L001 -p 8 -t ${library_tag[${SLURM_ARRAY_TASK_ID}]} \
      /n/groups/tabin/chlee/singlecell/FC_03554/Data/Intensities/BaseCalls/Undetermined_S0_L001_R2_001.fastq.gz \
      /n/groups/tabin/chlee/singlecell/FC_03554/Data/Intensities/BaseCalls/Undetermined_S0_L001_R4_001.fastq.gz \
      /n/groups/tabin/chlee/singlecell/FC_03554/Data/Intensities/BaseCalls/Undetermined_S0_L001_R1_001.fastq.gz \
      /n/groups/tabin/chlee/singlecell/FC_03554/Data/Intensities/BaseCalls/Undetermined_S0_L001_R3_001.fastq.gz 

But with this run, I see:

$ ls *.gz | head
dlib1.FC03554.L001.1.fastq.gz
dlib1.FC03554.L001.2.fastq.gz
dlib1.FC03554.L001.params.gz
dlib1.FC03554.L002.1.fastq.gz
dlib1.FC03554.L002.2.fastq.gz
dlib1.FC03554.L002.params.gz
dlib1.FC03554.L003.1.fastq.gz
dlib1.FC03554.L003.2.fastq.gz
dlib1.FC03554.L003.params.gz
dlib1.FC03554.L004.1.fastq.gz

The content is as follows:

$ zcat dlib1.FC03554.L001.params.gz | head
@QMJA4 CTGCGATGTCCCAATC CNNNNN AAAAAEE6AAAAA6/E E#####
@QMJA31 GTCCGTCACAGGAGGG CGNNNN AAAAAEEE6A/A/EEE E/####
@QMJA37 GGGGGGGGGGGGGGGG GGNGGN AAAAAEEEAAAA/AEE E/#E/#
@QMJA46 TCGAGTTTTTGGACTT CANCCN 6AA6A///6A//66<E EA#/E#
@QMJA47 GTACGGACAAATCGTT CANTGN AAAAA/EEAAAAAAEE EE#E<#
@QMJA60 CTTTATCCGCTGTCTG TTATAN A/AA/EEEAAA//E// 6EE//#
@QMJA66 CTGTTTCCGCTGTATC TATACN AAAAA6AEAAAAAEEE EEE<E#
@QMJA98 GGGGGGGGGGGGGGGG GGGGGG AAAAAEEEAAAAA//E /E/EEA
@QMJA102 GGGGGGGGGGGGGGGG GGGGGG AAAAAEEE///AA6/E /6/6E6
@QMJA113 AAATTCCGGGCATGCT GACCTA AAA//EAA//<AAEEE /A/EEE

**Comment by VPetukhov** [comment]
It must have created log file with prefix dlib${SLURM_ARRAY_TASK_ID}_FC03554 (you specified it with `-l`) and suffix tag_main.log. If I got your system right, dlib1.FC03554.L001.1.fastq.gz and dlib1.FC03554.L001.2.fastq.gz are exactly what was meant by .reads.gz. Am I misunderstood something?

**Comment by chlee-tabin** [comment]
Thank you for the clarification. I now realize that it is due to a bit ambiguity of the README.md document. In the document it states:


"To save this information, use -s option. In this case, in addition to fastq files with reads, the pipeline saves separate gzipped file with read parameters. This files must be passed to dropEst with -r option."

I thought "this files" are the .params.gz files, and these files are the ones to be passed with the -r option. 

If I understood you correctly, the -r option tag I should still nevertheless provide the fastq.gz files rather than the .params.gz files?

**Comment by chlee-tabin** [comment]
I have tested dropest and it appears that it recognizes the ".param.gz" file correctly, not the "fastq.gz" file. Also, it only worked with one file, and quoting filenames together separated by spaces did not work (produces an error message that it cannot find the file). Would it be possible to get some clarification how to use multiple files together?

**Comment by VPetukhov** [comment]
> If I understood you correctly, the -r option tag I should still nevertheless provide the fastq.gz files rather than the .params.gz files?

DropEst works with aligned .bam files, not .fast.gz files. I updated example commands in the readme:
```bash
dropest [options] [-m] [-r pipeline_res.params.gz] -g ./hg38/genes.gtf -c ./config.xml ./alignment.*/accepted_hits.bam
```

**Comment by VPetukhov** [comment]
> I have tested dropest and it appears that it recognizes the ".param.gz" file correctly, not the "fastq.gz" file. 

Please, post your command and program output here.

**Comment by chlee-tabin** [comment]
I think at this point, it was just my reading and partly a bit confusion of the wording of the README.md file. I would suggest that it would be helpful the dropest program help also reflects the correct (default) name of the parameter file and avoid potential confusion:

FROM

>-r, --read-params filenames: file or files with serialized params from tags search step. If there are several files, they should be provided in quotes, separated by space: "file1.reads.gz file2.reads.gz file3.reads.gz"

TO
>-r, --read-params filenames: file or files with serialized params from tags search step. If there are several files, they should be provided in quotes, separated by space: "file1.params.gz file2.params.gz file3.params.gz"

The only outstanding issue about this is that the -r option for dropEst was not accepting multiple files with quotes, but I just merged all .param.gz files together and supplied one file and worked fine.

The following is an example of the error I get when I supply multiple file names as advised per the `dropest --help` under the `-r` option:

```
cl266@compute-e-16-233:/n/groups/tabin/chlee/singlecell/FC_03726/drop$ ~/dropEst/dropest -m -V -r "dlib5.FC03554.L001.params.gz dlib5.FC03554.L002.params.gz" -g ../drop.reference.wo.pseudo/mm10.with.compact.transgenes.filtered.20180315.gtf -o dlib5.out/cell.counts.bcconfig.rds -c ~/dropEst/configs/indrop_v3_with_barcodes.xml dlib5.out/Aligned.out.sorted.bam 
/home/cl266/dropEst/dropest -m -V -r dlib5.FC03554.L001.params.gz dlib5.FC03554.L002.params.gz -g ../drop.reference.wo.pseudo/mm10.with.compact.transgenes.filtered.20180315.gtf -o dlib5.out/cell.counts.bcconfig.rds -c /home/cl266/dropEst/configs/indrop_v3_with_barcodes.xml dlib5.out/Aligned.out.sorted.bam 
Version: 0.8.0.
Can't open reads file 'dlib5.FC03554.L001.params.gz dlib5.FC03554.L002.params.gz'
cl266@compute-e-16-233:/n/groups/tabin/chlee/singlecell/FC_03726/drop$ ls dlib5.FC03554.L002.params.gz
dlib5.FC03554.L002.params.gz
cl266@compute-e-16-233:/n/groups/tabin/chlee/singlecell/FC_03726/drop$ ls dlib5.FC03554.L001.params.gz
dlib5.FC03554.L001.params.gz
```

**Comment by VPetukhov** [comment]
Thank you for suggestion! I made the correction you proposed (28c2f63c5d1ef958d0587fa1ee11974cfe8bb909).

**Comment by VPetukhov** [comment]
> The only outstanding issue about this is that the -r option for dropEst was not accepting multiple files with quotes, but I just merged all .param.gz files together and supplied one file and worked fine.

This is now fixed in b41b8c781ad7bbfa7002e2eabf7ada76e08892d9 commit.
